atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultShardProgressUpdaterTest.java

```diff
@@ -166,13 +217,69 @@ public void progressesPastOneOrMoreCompletedBucketsAndStopsCorrectly(
                                 + progressOnFinalBucket
                                         .map(BucketProgress::timestampProgress)
                                         .orElse(-1L));
+    }
+
+    @Test
+    public void doesNotLoadPastMaxBucketsForIteration() {
```

Comment:
Not necessary for this PR, but handles the case for more buckets than limit which existing tests don't (and surfaces the -1 / +1 error)